src/server/engine_shard_set.cc

```diff
@@ -133,10 +134,6 @@ class RoundRobinSharder {
   }
 
   static optional<ShardId> TryGetShardId(string_view key, XXH64_hash_t key_hash) {
-    if (!IsEnabled()) {
-      return nullopt;
-    }
-
```

Comment:
I think it's actually better to keep this here. It makes this function more resilient and does not require callers to check in advance.